### ISSUE TITLE
Implement basic infrastructure for optimization networks

### ISSUE THREAD

**Issue by HeuristicLab-Trac-Bot** [heal-research/HeuristicLab#2205]
*Issue migrated from trac ticket # 2205*

**milestone:** HeuristicLab 4.0 | **component:** Networks | **priority:** highest

#### 2014-07-01 01:56:00: @s-wagner created the issue

___
> **To Do:**
> * check handling of locked and read-only state of controls
> * executable networks
> 
> -*Topics for Discussion:**
> * design in general
>   - how much logic should be included in a port?
>   - problem of generic view types
>   - exception handling
>   - how to bridge the gap between internal and external ports of networks?
>   - declarative definition of results
> * port-to-port communication
>   - clone value?
>   - how to switch threads?
>   - should ports propagate their values immediately when they are connected?
> * data types
>   - POCOs vs. IItems
>   - generic vs. non-generic types (e.g. ports)
> * AlgorithmNode
>   - port updates due to result changes (speed, events)
> 
> This ticket has to be merged into stable after #1265
